datagateway_api/src/search_api/helpers.py

```diff
@@ -12,36 +21,176 @@
+                        entity_name, included_filter,
+                    ),
+                )
+
+    # Remove any duplicate ICAT relations
+    icat_relations = list(dict.fromkeys(icat_relations))
+    if icat_relations:
+        filters.append(PythonICATIncludeFilter(icat_relations))
+
     query = SearchAPIQuery(entity_name)
 
     filter_handler = FilterOrderHandler()
     filter_handler.add_filters(filters)
+    filter_handler.merge_python_icat_limit_skip_filters()
     filter_handler.apply_filters(query)
 
-    log.debug("Python ICAT Query: %s", query.icat_query.query)
-    # TODO - `getApiVersion()` used as a placeholder for testing client handling
-    # Replace with endpoint functionality when implementing the endpoints
-    return SessionHandler.client.getApiVersion()
+    log.debug("JPQL Query to be sent/executed in ICAT: %s", query.icat_query.query)
+    icat_query_data = query.icat_query.execute_query(SessionHandler.client, True)
+
+    panosc_data = []
+    for icat_data in icat_query_data:
+        panosc_model = getattr(models, entity_name)
+        panosc_record = panosc_model.from_icat(icat_data, icat_relations).json(
+            by_alias=True,
+        )
+        panosc_data.append(json.loads(panosc_record))
+
+    return panosc_data
 
 
 @client_manager
-def get_with_id(endpoint_name, entity_name, id_, filters):
-    pass
+def get_with_pid(entity_name, pid, filters):
+    """
+    Get a particular record of data from the specified entity
+
+    These will only be called with entity names of Dataset, Document and Instrument.
+    Each of these entities have a PID attribute, so we can assume the identifier will be
+    persistent (or `pid`) rather than an ordinary identifier (`id`)
+
+    :param entity_name: Name of the entity requested to query against
+    :type entity_name: :class:`str`
+    :param pid: Persistent identifier of the data to find
+    :type pid: :class:`str`
+    :param filters: The list of Search API filters to be applied to the request/query
+    :type filters: List of specific implementation :class:`QueryFilter`
+    :return: The (in JSON serialisable format) record of the specified PID
+    :raises MissingRecordError: If no results can be found for the query
+    """
+
+    log.info("Getting %s from ID %s", entity_name, pid)
+    log.debug("Entity Name: %s, Filters: %s", entity_name, filters)
+
+    filters.append(SearchAPIWhereFilter("pid", pid, "eq"))
+
+    query = SearchAPIQuery(entity_name)
+
+    filter_handler = FilterOrderHandler()
+    filter_handler.add_filters(filters)
+    filter_handler.merge_python_icat_limit_skip_filters()
+    filter_handler.apply_filters(query)
+
+    log.debug("JPQL Query to be sent/executed in ICAT: %s", query.icat_query.query)
+    icat_query_data = query.icat_query.execute_query(SessionHandler.client, True)
+
+    if not icat_query_data:
+        raise MissingRecordError("No result found")
+    else:
+        return icat_query_data[0]
```

Comment:
See my https://github.com/ral-facilities/datagateway-api/issues/311#issuecomment-1025857129

Reply:
Yeah that resolved it. Didn't get to fix the tests or push the fix so I'll get that done on Wednesday

Reply:
Fixed in 0de2b5b2b713699b66164ca5732888f997230aa5 and then I actually just call `get_search()` in 73b70bd756949fc23b7d4e58d6cdf5c3038a219b